File: java/src/main/java/com/deploykit/Plan.java
package com.deploykit;

import java.util.List;
import java.util.Map;

/** The rendered deployment plan for a single target. */
public final class Plan {
    private final String target;
    private final List<String> steps;
    private final Map<String, String> manifests;
    private final List<String> warnings;

    public Plan(String target, List<String> steps, Map<String, String> manifests, List<String> warnings) {
        this.target = target;
        this.steps = steps;
        this.manifests = manifests;
        this.warnings = warnings;
    }

    public String getTarget() {
        return target;
    }

    public List<String> getSteps() {
        return steps;
    }

    public Map<String, String> getManifests() {
        return manifests;
    }

    public List<String> getWarnings() {
        return warnings;
    }

    /** Reports whether the plan tripped no secure-by-default warnings. */
    public boolean isSecure() {
        return warnings.isEmpty();
    }
}

File: csharp/src/Plan.cs
using System.Collections.Generic;

namespace DeployKit;

/// <summary>The rendered deployment plan for a single target.</summary>
public sealed class Plan
{
    public string Target { get; }
    public IReadOnlyList<string> Steps { get; }
    public IReadOnlyDictionary<string, string> Manifests { get; }
    public IReadOnlyList<string> Warnings { get; }

    public Plan(
        string target,
        IReadOnlyList<string> steps,
        IReadOnlyDictionary<string, string> manifests,
        IReadOnlyList<string> warnings)
    {
        Target = target;
        Steps = steps;
        Manifests = manifests;
        Warnings = warnings;
    }

    /// <summary>Reports whether the plan tripped no secure-by-default warnings.</summary>
    public bool IsSecure() => Warnings.Count == 0;
}
